### PR TITLE
Binary writer

### PR DESCRIPTION
**Pull request by dtebbs**

### CODE REVIEW

File: src/main/java/io/BinaryCurveWriter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import algebra.fields.AbstractFieldElement;
+import algebra.fields.AbstractFieldElementExpanded;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+
+/**
+ * Base class for binary readers. Handles reading fixed-width BigInteger values (in big-endian
+ * form).
```
Comment (AntoineRondelet) [inline src/main/java/io/BinaryCurveWriter.java:14]: This comment seems to be a residual of copy/paste from the `BinaryCurveReader`.

`reader` -> `writer`, `reading` -> `writing`

File: src/test/java/zk_proof_systems/zkSNARK/grothBGM17/ZKSnarkObjectWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          FrT extends AbstractFieldElementExpanded<FrT>,
+          G1T extends AbstractG1<G1T>,
+          G2T extends AbstractG2<G2T>>
+      void writeProof(
+          final ZKSnarkObjectWriter<FrT, G1T, G2T> zkSnarkObjectWriter,
+          final FrT oneFr,
+          final G1T oneG1,
+          final G2T oneG2)
+          throws IOException {
+    final var dummyProof =
```
Comment (AntoineRondelet) [inline src/test/java/zk_proof_systems/zkSNARK/grothBGM17/ZKSnarkObjectWriterTest.java:45]: Why not use the function here. i.e. `dummyProof(oneFr, oneG1, oneG2),`?

File: src/main/java/io/BinaryCurveWriter.java (hunk 1 above)
Comment (dtebbs) [reply src/main/java/io/BinaryCurveWriter.java:14]: Updated the commebt.

File: src/test/java/zk_proof_systems/zkSNARK/grothBGM17/ZKSnarkObjectWriterTest.java (hunk 2 above)
Comment (dtebbs) [reply src/test/java/zk_proof_systems/zkSNARK/grothBGM17/ZKSnarkObjectWriterTest.java]: Switched to using the function.

**Review by AntoineRondelet — APPROVED** [review]
LGTM - thanks @dtebbs

**Pull request merged**